ThreeBotPackages/tfgrid/threebot_provisioning/chatflows/threebot_reservation.py

```diff
@@ -0,0 +1,135 @@
+from Jumpscale import j
+import netaddr
+import random
+import nacl
+import os
+
+
+def chat(bot):
+
+    """
+    This chat is create 3bot container using zosv2
+    """
+    explorer = j.clients.threebot.explorer
+    cl = j.clients.s3.get("deployer")
+
+    AWS_ID = cl.accesskey_
+    AWS_SECRET = cl.secretkey_
+
+    user_info = bot.user_info()
+    name = user_info["username"]
+    email = user_info["email"]
+    ips = ["IPV6", "IPV4"]
+    choose = ["New", "Restore"]
+    ip_range_choose = ["Specify IP Range", "Choose IP Range for me"]
+
+    folder_name = name.replace(".", "_")
+    env = dict()
+    secret_env = dict()
+
+    if not name or not email:
+        bot.md_show("Username or email not found in session. Please log in properly")
+
+    user_choice = bot.single_choice("create a new 3bot or restore existing 3bot?", choose)
+    identity = explorer.actors_all.phonebook.get(name=name, email=email)
+    identity_pubkey = identity.pubkey
+    if user_choice == "Restore":
+        password = bot.secret_ask("please add restore password?")
+        hash_restore = nacl.hash.blake2b(password.encode(), key=identity_pubkey.encode()).decode()
+
+    # ask user about corex user:password and ssh-key to give him full access to his container
+    pub_key = bot.string_ask(
+        "Please add your public ssh-key (that will allow you to access the deployed container using ssh) "
+    )
+    user_corex = bot.string_ask(
+        "username of your coreX (that will allow you to be secure when access from web browser)"
+    )
+    password = bot.secret_ask("password of your coreX (that will allow you to be secure when access from web browser)")
+
+    # create new reservation
+    reservation = j.sal.zosv2.reservation_create()
+
+    ip_version = bot.single_choice("choose your IP version", ips)
+    node_selected = j.sal.chatflow.nodes_get(1)[0]
+
+    # Encrypt AWS ID and AWS Secret to send it in secret env
+    aws_id_encrypt = j.sal.zosv2.container.encrypt_secret(node_selected.node_id, AWS_ID)
+    aws_secret_encrypt = j.sal.zosv2.container.encrypt_secret(node_selected.node_id, AWS_SECRET)
+
+    # Create network of reservation and add peers
+    if user_choice == "Restore":
+        hash_encrypt = j.sal.zosv2.container.encrypt_secret(node_selected.node_id, hash_restore)
+        env.update({"restore": "True"})
+        secret_env.update({"HASH": hash_encrypt})
+
+    reservation, config = j.sal.chatflow.network_configure(
+        bot, reservation, [node_selected], customer_tid=identity.id, ip_version=ip_version
+    )
+
+    ip_address = config["ip_addresses"][0]
+
+    backup = bot.single_choice("Auto-backup 3bot?", ["Yes", "No"])
```

Comment:
"Do you want your 3bot to be automatically backed up?"

and please move the logic for aws....

perhaps a private function will help make the code clearer